app/assets/javascripts/topology_preview.js

```diff
@@ -1,33 +1,34 @@
 var TopologyPreviewer = (function(){
-  var topologyGraph, svgGroup, data, root, tree, svgHeight, svgWidth;
+  var topologyGraph, presetData, style, svgGroup, data, root, tree, svgHeight, svgWidth,
+      currentStylesheet;
 
   var nodeIds         = 0,
       maxLabelLength  = 20,
-      duration        = 250,
-      endPointCounter = 0,
-      stepSize        = 20,
       ease            = 'cubic-out';
 
   var diagonal = d3.svg.diagonal()
-      .projection(function(d) { return [d.y, d.x]; });
+      .projection(function(d) { return [d.x, d.y]; });
 
   TopologyPreviewer.prototype = {
     preview: function(){
       clear();
 
-      svgHeight = viewerHeight();
-      svgWidth  = viewerWidth();
-      svg       = buildBaseSvg();
-      data      = [graphData()];
-      root      = data[0];
+      currentStylesheet = TopologyStylesheet[style];
```

Comment:
What I mean is, if I do:

```js
new TopologyPreviewer(...)
new TopologyPreviewer(...)
```

... each instance doesn't get it's own variables; won't the vars used by the second Previewer overwrite the vars of the first? Putting them inside a closure means they're not globally *readable*, but still only one can exist at any moment in time, right? Maybe I'm misunderstanding?